packages/react/ds-app-launchpad/src/ui/GitDiffViewer/Provider.tsx

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useState } from "react";
+import Context from "./Context.js";
+import "./styles.css";
+import type { ProviderOptions } from "./types.js";
+
+const componentCssClassName = "ds git-diff-viewer";
+
+const Provider = ({
+  id,
+  className,
+  style,
+  children,
+  ...contextOptions
```

Comment:
it is easier to maintain the logic of all or none type if we forward the options to the context.